### PR TITLE
fix(checkbox): Set aria-checked to mixed for indeterminate checkbox

### LINKED ISSUE #7932: checkbox a11y issue: Partially done task is getting read out as 'Checkbox not checked'.
#### Bug, feature request, or proposal:

Bug

#### What is the expected behavior?
Partially done task should read out as 'indeterminate'.

#### What is the current behavior?

Partially done task is getting read out as 'Checkbox not checked'. It might cause confusion to end user. 

#### What are the steps to reproduce?
1. Click on Checkbox.2. Navigate over nested checklist.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

Chromevox Chrome OS Version 61.0.3163.100 (Official Build) (64-bit) , Voiceover (mac OS version: 10.12.6)

### PR DESCRIPTION
**Pull request by tinayuangao**
Fixes #7932

### CODE REVIEW

File: src/lib/checkbox/checkbox.ts [hunk 1]
```diff
@@ -303,6 +303,10 @@ export class MatCheckbox extends _MatCheckboxMixinBase implements ControlValueAc
     this._changeDetectorRef.markForCheck();
   }
 
+  getAriaChecked() {
+    return this.checked ? true : (this.indeterminate ? 'mixed' : false);
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.ts]: Return type should be `'true' | 'false' | 'mixed'` (since the attr should always be a string)

Comment (tinayuangao) [reply src/lib/checkbox/checkbox.ts]: Done. Thanks for review!

File: src/lib/checkbox/checkbox.ts [hunk 2]
```diff
@@ -303,6 +303,10 @@ export class MatCheckbox extends _MatCheckboxMixinBase implements ControlValueAc
     this._changeDetectorRef.markForCheck();
   }
 
+  getAriaChecked(): string {
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.ts]: Just also realized: method should be internal `_getAriaChecked`

The return type can really the string literal type `'true' | 'false' | 'mixed'`

File: src/lib/checkbox/checkbox.spec.ts [hunk 3]
```diff
@@ -87,13 +87,17 @@ describe('MatCheckbox', () => {
       expect(checkboxNativeElement.classList).not.toContain('mat-checkbox-checked');
       expect(inputElement.checked).toBe(false);
       expect(inputElement.indeterminate).toBe(false);
+      expect(inputElement.getAttribute('aria-checked'))
+        .toBe('false', 'Expect aria-checked to be false');
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.spec.ts]: +2 more indent

File: src/lib/checkbox/checkbox.ts [hunk 4]
```diff
@@ -303,6 +303,10 @@ export class MatCheckbox extends _MatCheckboxMixinBase implements ControlValueAc
     this._changeDetectorRef.markForCheck();
   }
 
+  _getAriaChecked(): 'true'|'false'|'mixed' {
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.ts]: We typically put spaces between the types like this
```ts
'true' | 'false' | 'mixed'
```

**Pull request merged**